docs/schema-concept/circular-references-and-dynamic-types.md

```diff
@@ -2,48 +2,124 @@
-    type: 'Book', // << dynamic type name
-    required: true
+ */
+const User = attributes(
+  {
+    name: String,
+    friends: {
+      type: Array,
+      itemType: 'User', // << dynamic type with inferred identifier
+    },
+    favoriteBook: {
+      type: 'BookStructure', // << dynamic type with custom identifier
+      required: true,
+    },
+    books: {
+      type: 'BooksCollection', // << dynamic type with inferred identifier
+      itemType: String,
+    },
   },
-  books: {
-    type: 'BooksCollection', // << dynamic type name
-    itemType: String
+  {
+    dynamics: {
+      /* dynamic types */
+      User: () => User,
+      BookStructure: () => require('./Book'),
+      BooksCollection: () => require('./BooksCollection'),
+    },
   }
-}, {
-  dynamics: { // << dynamic types values
-    User: () => User,
-    Book: () => require('./Book'),
-    BooksCollection: () => require('./BooksCollection')
-  }
-})(class User { });
+)(class User {});
 
 /*
  * Book.js
-*/
-const Book = attributes({
-  name: String,
-  owner: 'User', // << dynamic type name
-  nextBook: 'Book' // << dynamic type name
-}, {
-  dynamics: { // << dynamic types values
-    User: () => require('./User'),
-    Book: () => Book
+ */
+const Book = attributes(
+  {
+    name: String,
+    owner: 'User', // << dynamic type with inferred identifier
+    nextBook: 'BookStructure', // << dynamic type with custom identifier
+  },
+  {
+    identifier: 'BookStructure', // << custom identifier
+    dynamics: {
+      /* dynamic types */
+      User: () => require('./User'),
+      BookStructure: () => Book,
+    },
+  }
+)(class Book {});
+```
+
+## Dynamic type identifier
+
+The identifier of the type has to be same everywhere it's used, and can be defined in two ways:
+
+### Inferred identifier
+
+The identifier can be inferred based in the class that is wrapped by the `attributes` function. In backend scenarios this will be the most common case:
+
+```js
+const User = attributes(
+  {
+    name: String,
+    bestFriend: 'User', // [A] type with inferred identifier
+  },
+  {
+    dynamics: {
+      User: () => User, // [B] inferred identifier concrete type
+    },
+  }
+)(
+  class User {
+    //   ^ the name of this class is the identifier
+    // so if we change this name to UserEntity, we'll have to change
+    // both [A] and [B] to use the string 'UserEntity' instead of 'User'
   }
-})(class Book { });
+);
 ```
 
-Notice that the _name_ of the type has to be the same name of the key inside the `dynamics` object. Also, it's important that the require be done __inside__ the function when the dynamic type also references the current type (in the example above, the type `User` has an attribute of the type `Book`, and the type `Book` has an attribute of the type `User`).
+### Custom identifier
+
+If for some reason you can't rely in the class name, be it because you're using a compiler that strip class names or create a dynamic one, you can explicitly set one.
```

Comment:
```suggestion
If for some reason you can't rely on the class name, be it because you're using a compiler that strip class names or create a dynamic one, you can explicitly set one.
```

you rely on/upon something :ok_woman: